docs/icicle/primitives/msm.md

```diff
@@ -1,19 +1,88 @@
-# MSM
+# MSM - Multi scalar multiplication
 
-MSM (Multi scalar multiplication) is an important primitive in ZK protocols.
+MSM stands for Multi scalar multiplication, its defined as:
 
-### Using MSM
+<math xmlns="http://www.w3.org/1998/Math/MathML">
+  <mi>M</mi>
+  <mi>S</mi>
+  <mi>M</mi>
+  <mo stretchy="false">(</mo>
+  <mi>a</mi>
+  <mo>,</mo>
+  <mi>G</mi>
+  <mo stretchy="false">)</mo>
+  <mo>=</mo>
+  <munderover>
+    <mo data-mjx-texclass="OP" movablelimits="false">&#x2211;</mo>
+    <mrow data-mjx-texclass="ORD">
+      <mi>j</mi>
+      <mo>=</mo>
+      <mn>0</mn>
+    </mrow>
+    <mrow data-mjx-texclass="ORD">
+      <mi>n</mi>
+      <mo>&#x2212;</mo>
+      <mn>1</mn>
+    </mrow>
+  </munderover>
+  <msub>
+    <mi>a</mi>
+    <mi>j</mi>
+  </msub>
+  <msub>
+    <mi>G</mi>
+    <mi>j</mi>
+  </msub>
+</math>
 
-#### Supported algorithms
+Where
 
-Our MSM implementation supports two algorithms.
+$G_j \in G$ - points from an Elliptic Curve group.
 
--   Bucket accumulation 
--   Large triangle accumulation
```

Comment:
@DmytroTym Please verify my details on the `Large triangle accumulation` since im more familiar with the `Bucket accumulation` method